zetta_utils/db_annotations/precomp_annotations.py

```diff
@@ -328,22 +329,45 @@ def subdivide(data, bounds: VolumetricIndex, chunk_sizes, write_to_dir=None, lev
         grid_shape = ceil(bounds_size / chunk_size)
         logger.info(f"subdividing {bounds} by {chunk_size}, for grid_shape {grid_shape}")
         level_key = f"spatial{level}"
-        for x, y, z in product(range(grid_shape[0]), range(grid_shape[1]), range(grid_shape[2])):
-            chunk_start = bounds.start + Vec3D(x, y, z) * chunk_size
-            chunk_end = chunk_start + chunk_size
-            chunk_bounds = VolumetricIndex.from_coords(chunk_start, chunk_end, bounds.resolution)
-            # pylint: disable=cell-var-from-loop
-            chunk_data: Sequence[LineAnnotation] = list(
-                filter(lambda d: d.in_bounds(chunk_bounds), data)
+        # total_qty = grid_shape[0] * grid_shape[1] * grid_shape[2]
+        qty_done = 0
+        for x in range(grid_shape[0]):
+            x_start = bounds.start[0] + x * chunk_size[0]
+            x_end = x_start + chunk_size[0]
+            x_idx = VolumetricIndex.from_coords(
+                (x_start, bounds.start[1], bounds.start[2]),
+                (x_end, bounds.stop[1], bounds.stop[2]),
+                bounds.resolution,
             )
-            # logger.info(f'spatial{level}/{x}_{y}_{z} contains {len(chunk_data)} lines')
-            limit = max(limit, len(chunk_data))
-            if write_to_dir is not None and level in levels_to_write:
-                level_dir = path_join(write_to_dir, level_key)
-                if not os.path.exists(level_dir):
-                    os.makedirs(level_dir)
-                anno_file_path = path_join(level_dir, f"{x}_{y}_{z}")
-                write_lines(anno_file_path, chunk_data)
+            data_within_x = list(filter(lambda d: d.in_bounds(x_idx), data))
+            for y in range(grid_shape[1]):
+                y_start = bounds.start[1] + y * chunk_size[1]
+                y_end = y_start + chunk_size[1]
+                y_idx = VolumetricIndex.from_coords(
+                    (x_start, y_start, bounds.start[2]),
+                    (x_end, y_end, bounds.stop[2]),
+                    bounds.resolution,
+                )
+                data_within_xy = list(filter(lambda d: d.in_bounds(y_idx), data_within_x))
+                for z in range(grid_shape[2]):
+                    qty_done += 1
+                    chunk_start = bounds.start + Vec3D(x, y, z) * chunk_size
+                    chunk_end = chunk_start + chunk_size
+                    chunk_bounds = VolumetricIndex.from_coords(
+                        chunk_start, chunk_end, bounds.resolution
+                    )
+                    # pylint: disable=cell-var-from-loop
+                    chunk_data: Sequence[LineAnnotation] = list(
+                        filter(lambda d: d.in_bounds(chunk_bounds), data_within_xy)
```

Comment:
I think passing chunk_bounds as default parameter might silence pylint. Not sure if that's cleaner, though
```suggestion
                        filter(lambda d, cb=chunk_bounds: d.in_bounds(cb), data_within_xy)
```

Reply:
Oh actually I tried this, and mypy doesn't like it.  So I'm going to go back to the pylint disable approach.